test/cpp/api/functional.cpp

```diff
@@ -360,3 +360,24 @@ TEST_F(FunctionalTest, LogSigmoid) {
   auto y_exp = torch::log(torch::ones_like(x)/(torch::ones_like(x) + torch::exp(torch::neg(x))));
   ASSERT_TRUE(torch::allclose(y, y_exp, 1e-4, 1e-7));
 }
+
+TEST_F(FunctionalTest, Normalize) {
+  {
+    auto input = torch::randn({1,3,4,4}, torch::requires_grad());
+    auto norm = F::normalize(input, NormalizeOptions().p(1).dim(-1));
+    // reduce to scalar to call .backward()
+    torch::Tensor s = norm.sum();
+    s.backward();
+    ASSERT_EQ(s.ndimension(), 0);
+  }
+  {
+    auto input = torch::randn({1,3,4,4});
+    auto output = torch::randn({1,3,4,4});
+    F::normalize(input, NormalizeOptions().dim(-2), output);
```

Comment:
It would be great to have specific values for `input` (e.g. `torch::arange(48).reshape({1, 3, 4, 4})`), and then check that the value of `output` is what we expect it should be.

Reply:
Yes this was my first intuition as well but I decided to follow the python tests as is. Will make the necessary changes and maybe in the python tests in a future pr